### ISSUE TITLE
EAP2024.1 startup crash

### ISSUE THREAD

**Issue by jerzyk** [trinhanhngoc/pycharm-odoo#455]
can't start EAP2024.1 (today was an second update, but error was as well with previous build), need to manually remove Odoo plugin to even start PyCharm

I suppose it is related to IndexNotReady as here too are calls to openapi?

but after removing Odoo plugin IDE is working fine

```
Internal error. Please refer to https://jb.gg/ide/critical-startup-errors

java.lang.AssertionError: kotlinx.coroutines.JobCancellationException: Parent job is Cancelling; job=UndispatchedCoroutine{Cancelling}@7ff2a664
    at com.intellij.util.EnvironmentUtil.getEnvironmentMap(EnvironmentUtil.java:98)
    at com.intellij.util.EnvironmentUtil.getValue(EnvironmentUtil.java:192)
    at com.intellij.ide.j.E.Nh.N(Nh.java:146)
    at com.intellij.ide.j.E.jT.p(jT.java:213)
    at com.intellij.ide.j.E.e.N(e.java:201)
    at com.intellij.ide.j.E.s.N(s.java:341)
    at com.intellij.ide.j.E.s.N(s.java:203)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
    at com.intellij.idea.j.invokeSuspend(j.java:59)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
    at com.intellij.idea.j.N(j.java)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
    at com.intellij.idea.j.invoke(j.java)
    at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
    at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
    at com.intellij.idea.MainImpl.start(MainImpl.kt:58)
    at com.intellij.platform.ide.bootstrap.StartupUtil$startApplication$8$2.invokeSuspend(main.kt:294)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
    at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
    at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
    at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
    at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
    at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
    at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
    at com.intellij.idea.Main.mainImpl(Main.kt:64)
    at com.intellij.idea.Main.mainImpl$default(Main.kt:50)
    at com.intellij.idea.Main.main(Main.kt:47)
    Suppressed: java.lang.NoSuchMethodError: 'com.intellij.openapi.components.ComponentManager com.intellij.openapi.extensions.impl.ExtensionPointImpl.getComponentManager()'
        at dev.ngocta.pycharm.odoo.OdooApplicationInitializedListener.adjustCompletionContributor(OdooApplicationInitializedListener.java:21)
        at dev.ngocta.pycharm.odoo.OdooApplicationInitializedListener.componentsInitialized(OdooApplicationInitializedListener.java:14)
        at com.intellij.ide.ApplicationInitializedListener$execute$2.invoke(ApplicationInitializedListener.kt:36)
        at com.intellij.ide.ApplicationInitializedListener$execute$2.invoke(ApplicationInitializedListener.kt:34)
        at com.intellij.openapi.progress.CoroutinesKt.blockingContextInner(coroutines.kt:320)
        at com.intellij.openapi.progress.CoroutinesKt.access$blockingContextInner(coroutines.kt:1)
        at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invokeSuspend(coroutines.kt:197)
        at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
        at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
        at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
        at com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:196)
        at com.intellij.ide.ApplicationInitializedListener.execute$suspendImpl(ApplicationInitializedListener.kt:34)
        at com.intellij.ide.ApplicationInitializedListener.execute(ApplicationInitializedListener.kt)
        at com.intellij.platform.ide.bootstrap.ApplicationLoader$callAppInitialized$1.invokeSuspend(ApplicationLoader.kt:495)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
Caused by: kotlinx.coroutines.JobCancellationException: Parent job is Cancelling; job=UndispatchedCoroutine{Cancelled}@7ff2a664
Caused by: [CIRCULAR REFERENCE: java.lang.NoSuchMethodError: 'com.intellij.openapi.components.ComponentManager com.intellij.openapi.extensions.impl.ExtensionPointImpl.getComponentManager()']

```

**Comment by trinhanhngoc** [comment]
Hello @jerzyk ,

I will release an update that supports PyCharm 2024.1 EAP this week.

**Comment by trinhanhngoc** [comment]
Hello @jerzyk ,

The new version 2024.1.2 has been released with support for PyCharm 2024.1.

**Comment by jerzyk** [comment]
Thanks, will check it shortly
